pkg/question/helpers_test.go

```diff
@@ -0,0 +1,67 @@
+package question_test
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/core"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/stretchr/testify/assert"
+)
+
+type QA struct {
+	prompt survey.Prompt
+	answer any
+	err    error
+}
+
+func (qa *QA) String() string {
+	return fmt.Sprintf("{\n"+
+		"	prompt: %+v\n"+
+		"	answer: %v\n"+
+		"	err: %v"+
+		"\n}\n", qa.prompt, qa.answer, qa.err)
+}
+
+func NewAskMocker(t assert.TestingT, questions []QA) (question.Asker, func()) {
+	c := make(chan QA)
```

Comment:
What's the channel doing here? Do we have any actual concurrency?

It appears as though you're front-loading all the questions into the channel's internal buffer then pulling one item out each time a question gets asked. If that's the case then that appears to be equivalent to a plain old array-based queue; you could simply pop the first item off an array of questions, couldn't you?

e.g line 86 here:
https://github.com/OctopusDeploy/cli/pull/27/files#diff-969618f76259350f331567e47f400bdf86b86f0e4c5b95989df775b9195415d5R86

Reply:
Yes, having an array that dequeues could work here. I decided to use a generator pattern. The channel does not actually front-load, sorta. `c <- qa`  is sorta like yield, it will wait for the channel to be read before looping again. Agreed this is probably "over engineered" but It works so I will leave it as is for now.

Reply:
Yeah I didn't block the PR on behalf of this, but it is a bit weird; Seeing channels in the code signals "concurrency" which is a false signal here